### ISSUE TITLE
'login=' is an invalid login.

### ISSUE THREAD

**Issue by b4iterdev** [ddclient/ddclient#479]
Hi, I'm trying to setup ddclient and I get this error: 
Setting up watches.
Watches established.
```
WARNING:  skipping host: b4iterdev.tk: 'login=' is an invalid login.
WARNING:  skipping host: \: 'login=' is an invalid login.
WARNING:  skipping host: host.b4iterdev.tk: 'login=' is an invalid login.
WARNING:  skipping host: \: 'login=' is an invalid login.
WARNING:  skipping host: b4iterdev.tk: 'login=' is an invalid login.
WARNING:  skipping host: host.b4iterdev.tk: 'login=' is an invalid login.
```
here's my config: 

```
daemon=300				# check every 300 seconds
syslog=yes				# log update msgs to syslog
#mail=root				# mail all msgs to root
#mail-failure=root	    # mail failed update msgs to root
pid=/var/run/ddclient/ddclient.pid	# record PID in file.
ssl=yes 				# use ssl-support.  Works with
						# ssl-library
use=web, web=checkip.dyndns.org/, web-skip='Current IP Address: ' # found after IP Address

##
## CloudFlare (www.cloudflare.com)
##
protocol=cloudflare,        \
login=token,     \ # Only needed if you are using your global API key. If you are using an API token, set it to "token" (wihtout double quotes).
password=[token],             \ # This is either your global API key, or an API token. If you are using an API token, it must have the permissions "Zone - DNS - Edit" and "Zone - Zone - Read". The Zone resources must be "Include - All zones".
zone=b4iterdev.tk,            \
ttl=1,                      \
b4iterdev.tk,host.b4iterdev.tk

```
I tried to put login and password under ttl line but still no luck, how can I solve this, Thanks!

**Comment by SuperSandro2000** [comment]
I don't know.

**Comment by hugeblank** [comment]
You don't know, so the solution is to close the issue? There's clearly an issue with either the config documentation or ddclient's handling of cloudflare API tokens. This should be reopened and a closer look should be had.

**Comment by b4iterdev** [comment]
Well I did not close this issue, but I use DNS-O-Matic as an alternative solution

**Comment by amallik123** [comment]
I am adding this just in case someone else comes across this, the problem appears to be the extra , and / in the cloudflare example config, in other words, just change it to something like this by removing all the commas and slashes (worked for me):

protocol=cloudflare
login=token
password=[token]
zone=b4iterdev.tk
ttl=1
b4iterdev.tk,host.b4iterdev.tk

**Comment by gleason21** [comment]
> I am adding this just in case someone else comes across this, the problem appears to be the extra , and / in the cloudflare example config, in other words, just change it to something like this by removing all the commas and slashes (worked for me):
> 
> protocol=cloudflare login=token password=[token] zone=b4iterdev.tk ttl=1 b4iterdev.tk,host.b4iterdev.tk

This solution works (tho I kept the commas in) and still seems to be an issue in latest version (3.11.2). (Also SuperSandro2000 is a dogshit developer for closing this issue out :) )

**Comment by rhansen** [comment]
This was fixed by commit 3c522a7aa235f63ae0439e5674e7406e20c90956, which will be in the upcoming v4.0.0 release.

**Comment by hugeblank** [comment]
Thanks for the confirmation that the issue is fixed.

> GitHub UI makes it way too easy to accidentally close a bug report when commenting, which I'm assuming is what happened here.

Best case, yeah, but also "I don't know" is not all that useful of a comment in the first place making the whole situation a gear grinder regardless. The comment flagged as abuse is totally justified though, jeez.
